### PR TITLE
Check enum values before pass to native side.

### PR DESCRIPTION
**Pull request by hwanseung**
Check enum values before pass to native side.

ISSUE=#80

### CODE REVIEW

**Comment by hwanseung** [comment]
@romandev 
this PR is same with initial #124.

**Comment by hwanseung** [comment]
i rebased this patch.

**Comment by romandev** [comment]
@hwanseung Your patch is broken in bots :(

**Comment by hwanseung** [comment]
@romandev i fixed it.

File: template/enum_types_header.njk [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * limitations under the License.
+ */
+
+#ifndef GEN_ENUM_TYPES_H_
+#define GEN_ENUM_TYPES_H_
+
+#include <set>
+#include <string>
+
+class EnumTypes {
```
Comment (romandev) [inline template/enum_types_header.njk]: IMHO, we don't need to generate this type.
We can just pass the string set into isValidEnum() directly.

File: generator/main.ts [hunk 2]
```diff
@@ -72,6 +74,52 @@ async function generateInterface(
...
+
+  await file.write(
+      enum_type_header_path,
+      env.renderString(enum_type_header_tmpl, {enums: idl_enum}));
+  await file.write(
+      enum_type_cpp_path,
+      env.renderString(enum_type_cpp_tmpl, {enums: idl_enum}));
+}
+
+// TODO(hwansueng) :: this function should be improved.
```
Comment (romandev) [inline generator/main.ts]: nit: // TODO(hwanseung): This function should be improved.

**Review by romandev — COMMENTED** [review]
Looks good overall.
I think we don't need to generate enum_types_cpp(.h, .cc).

File: template/enum_types_header.njk (hunk 1 above)
Comment (hwanseung) [reply template/enum_types_header.njk]: do you means to pass the string set into isVaildEnum() directly in interface_cpp.njk?

Comment (romandev) [reply template/enum_types_header.njk]: Yeah, exactly!

File: generator/main.ts (hunk 2 above)
Comment (hwanseung) [reply generator/main.ts]: done

File: template/enum_types_header.njk (hunk 1 above)
Comment (hwanseung) [reply template/enum_types_header.njk]: i changed to pass to the string set as parameter.

**Pull request merged**